tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/Serverless.cs

```diff
@@ -0,0 +1,95 @@
+// <copyright file="Serverless.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.IO;
+
+using Datadog.Trace.Logging;
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.ClrProfiler.ServerlessInstrumentation
+{
+    internal static class Serverless
+    {
+        private const string DefinitionsId = "68224F20D001430F9400668DD25245BA";
+        private const string ExtensionEnvName = "_DD_EXTENSION_PATH";
+        private const string ExtensionFullPath = "/opt/extensions/datadog-agent";
+        private const string FunctionEnvame = "AWS_LAMBDA_FUNCTION_NAME";
+        private const string HandlerEnvName = "_HANDLER";
+        private const string LogLevelEnvName = "DD_LOG_LEVEL";
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(Serverless));
+
+        internal static void InitIfNeeded()
+        {
+            if (IsRunningInLambda(ExtensionFullPath))
+            {
+                Debug("Sending CallTarget serverless integration definitions to native library.");
+                var serverlessDefinitions = GetServerlessDefinitions();
+                NativeMethods.InitializeProfiler(DefinitionsId, serverlessDefinitions);
+                foreach (var def in serverlessDefinitions)
+                {
+                    def.Dispose();
+                }
+
+                Debug("The profiler has been initialized with serverless definitions, count = " + serverlessDefinitions.Length);
+            }
+        }
+
+        internal static bool IsRunningInLambda(string extensionPath)
+        {
+            string path = EnvironmentHelpers.GetEnvironmentVariable(ExtensionEnvName) ?? extensionPath;
+            return File.Exists(path) && !string.IsNullOrEmpty(EnvironmentHelpers.GetEnvironmentVariable(FunctionEnvame));
+        }
+
+        internal static NativeCallTargetDefinition[] GetServerlessDefinitions()
+        {
+            try
+            {
+                LambdaHandler handler = new LambdaHandler(EnvironmentHelpers.GetEnvironmentVariable(HandlerEnvName));
+                string assymblyName = typeof(InstrumentationDefinitions).Assembly.FullName;
+                string integrationType = GetIntegrationTypeFromParamCount(handler.ParamTypeArray.Length);
+                return new NativeCallTargetDefinition[]
+                {
+                    new(handler.GetAssembly(), handler.GetFullType(), handler.GetMethodName(), handler.ParamTypeArray, 0, 0, 0, 65535, 65535, 65535, assymblyName, integrationType)
```

Comment:
just a little typo 🙂
```suggestion
                string assemblyName = typeof(InstrumentationDefinitions).Assembly.FullName;
                string integrationType = GetIntegrationTypeFromParamCount(handler.ParamTypeArray.Length);
                return new NativeCallTargetDefinition[]
                {
                    new(handler.GetAssembly(), handler.GetFullType(), handler.GetMethodName(), handler.ParamTypeArray, 0, 0, 0, 65535, 65535, 65535, assemblyName, integrationType)
```